### ISSUE TITLE
[Bug]: FormatException: Invalid Zip Signature

### ISSUE THREAD

**Issue by lengerke** [LinwoodDev/Butterfly#759]
### What happened?

I cannot open one specific file anymore that I worked on. Before the file was corrupted, there was a significant delay between the stylo touching the display and the app drawing the corresponding line (up to several seconds). The tablet was otherwise responsive and I worked offline. I don't know if that is relevant, though.

### Version

2.2.1

### Platform

Android

### Relevant log output

```shell
#0      new ZipFile (package:archive/src/zip/zip_file.dart:52)
#1      ZipFileHeader.readLocalFileHeader (package:archive/src/zip/zip_file_header.dart:104)
#2      new ZipDirectory.read (package:archive/src/zip/zip_directory.dart:63)
#3      ZipDecoder.decodeBuffer (package:archive/src/zip_decoder.dart:24)
#4      ZipDecoder.decodeBytes (package:archive/src/zip_decoder.dart:16)
#5      noteDataMigrator (package:butterfly_api/src/converter/note.dart:16)
#6      new NoteData.fromData (package:butterfly_api/src/models/data.dart:29)
#7      _decode (package:butterfly/api/file_system.dart:16)
#8      TypedDirectoryFileSystem._toTypedAsset (package:lw_file_system/src/api/typed.dart:78)
#9      TypedDirectoryFileSystem.readAsset.<anonymous closure> (package:lw_file_system/src/api/typed.dart:117)
<asynchronous suspension>
#10     GeneralDirectoryFileSystem.fetchAsset (package:lw_file_system/src/api/base/directory.dart:25)
<asynchronous suspension>
#11     Stream.last.<anonymous closure> (dart:async/stream.dart:1607)
<asynchronous suspension>
#12     _ProjectPageState._load (package:butterfly/views/main.dart:210)
<asynchronous suspension>
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by CodeDoctorDE** [comment]
Hmm, looks like a duplicate of #746 but I really don't know how to reproduce it.
Looks like a bug inside the zip library that I use. I'm always happy for contributions or reproducible steps that I can test.

**Comment by CodeDoctorDE** [comment]
Did you have used any "non-ascii" names? Maybe this could be the problem: https://github.com/brendan-duncan/archive/issues/354

**Comment by CodeDoctorDE** [comment]
Should be fixed in 2.2.4, exisitng files are sadly not working
